File: CS/WindowsApplication3/Main.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using System.Data.OleDb;
using System.Threading;
using System.Runtime.Remoting.Messaging;
using DevExpress.XtraGrid.Columns;
using DevExpress.Skins;
using DevExpress.LookAndFeel;
using DXSample.Properties;


namespace DXSample {
    public partial class Main: XtraForm {
        UnboundColumnHelper _largeImageHelper;

        public Main() {
            InitializeComponent();
            InitializeUnboundLargeImageColumn();
            gridView1.CustomRowCellEdit += new DevExpress.XtraGrid.Views.Grid.CustomRowCellEditEventHandler(gridView1_CustomRowCellEdit);
            _largeImageHelper.AsyncOperationIsComplete += new EventHandler(_largeImageHelper_AsyncOperationIsComplete);
            gridView1.CustomRowCellEditForEditing += new DevExpress.XtraGrid.Views.Grid.CustomRowCellEditEventHandler(gridView1_CustomRowCellEditForEditing);
            gridView1.OptionsView.AnimationType = DevExpress.XtraGrid.Views.Base.GridAnimationType.AnimateAllContent;
        }

        void gridView1_CustomRowCellEditForEditing(object sender, DevExpress.XtraGrid.Views.Grid.CustomRowCellEditEventArgs e) {
            if(e.Column.FieldName == "LargeImage") {
                e.RepositoryItem = new DevExpress.XtraEditors.Repository.RepositoryItemPictureEdit();
            }
        }

        void _largeImageHelper_AsyncOperationIsComplete(object sender, EventArgs e) {
            UpdateGrid();
        }

        public void UpdateGrid() {
            if(gridControl1.InvokeRequired)
                gridControl1.Invoke(new MethodInvoker(delegate {
                    gridView1.LayoutChanged();
                }));
            else
                gridView1.LayoutChanged();
        }

        private void InitializeUnboundLargeImageColumn() {
            GridColumn unbColumn = gridView1.Columns.AddField("LargeImage");
            unbColumn.VisibleIndex = gridView1.Columns.Count;
            unbColumn.UnboundType = DevExpress.Data.UnboundColumnType.Object;
            // Disable editing.
            unbColumn.OptionsColumn.AllowEdit = false;
            // Specify format settings.
            unbColumn.DisplayFormat.FormatType = DevExpress.Utils.FormatType.Custom;
            // Customize the appearance settings.
            unbColumn.AppearanceCell.BackColor = Color.LemonChiffon;
            // Assign to helper class
            _largeImageHelper = new UnboundColumnHelper(unbColumn);
        }

        void gridView1_CustomRowCellEdit(object sender, DevExpress.XtraGrid.Views.Grid.CustomRowCellEditEventArgs e) {
            if(e.Column.FieldName == "LargeImage") {
                e.RepositoryItem = new DevExpress.XtraEditors.Repository.RepositoryItemPictureEdit();
            }
        }

        private void OnFormLoad(object sender, EventArgs e) {
            this.ordersTableAdapter.Fill(this.nwindDataSet.Orders);
        }
    }
}

File: VB/WindowsApplication3/Main.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports System.Data.OleDb
Imports System.Threading
Imports System.Runtime.Remoting.Messaging
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.Skins
Imports DevExpress.LookAndFeel
Imports DXSample.Properties


Namespace DXSample
    Partial Public Class Main
        Inherits XtraForm
        Private _largeImageHelper As UnboundColumnHelper

        Public Sub New()
            InitializeComponent()
            InitializeUnboundLargeImageColumn()
            AddHandler gridView1.CustomRowCellEdit, AddressOf gridView1_CustomRowCellEdit
            AddHandler _largeImageHelper.AsyncOperationIsComplete, AddressOf _largeImageHelper_AsyncOperationIsComplete
            AddHandler gridView1.CustomRowCellEditForEditing, AddressOf gridView1_CustomRowCellEditForEditing
            gridView1.OptionsView.AnimationType = DevExpress.XtraGrid.Views.Base.GridAnimationType.AnimateAllContent
        End Sub

        Private Sub gridView1_CustomRowCellEditForEditing(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Grid.CustomRowCellEditEventArgs)
            If e.Column.FieldName = "LargeImage" Then
                e.RepositoryItem = New DevExpress.XtraEditors.Repository.RepositoryItemPictureEdit()
            End If
        End Sub

        Private Sub _largeImageHelper_AsyncOperationIsComplete(ByVal sender As Object, ByVal e As EventArgs)
            UpdateGrid()
        End Sub

        Public Sub UpdateGrid()
            If gridControl1.InvokeRequired Then
                gridControl1.Invoke(New MethodInvoker(Function() AnonymousMethod1()))
            Else
                gridView1.LayoutChanged()
            End If
        End Sub

        Private Function AnonymousMethod1() As Boolean
            gridView1.LayoutChanged()
            Return True
        End Function

        Private Sub InitializeUnboundLargeImageColumn()
            Dim unbColumn As GridColumn = gridView1.Columns.AddField("LargeImage")
            unbColumn.VisibleIndex = gridView1.Columns.Count
            unbColumn.UnboundType = DevExpress.Data.UnboundColumnType.Object
            ' Disable editing.
            unbColumn.OptionsColumn.AllowEdit = False
            ' Specify format settings.
            unbColumn.DisplayFormat.FormatType = DevExpress.Utils.FormatType.Custom
            ' Customize the appearance settings.
            unbColumn.AppearanceCell.BackColor = Color.LemonChiffon
            ' Assign to helper class
            _largeImageHelper = New UnboundColumnHelper(unbColumn)
        End Sub

        Private Sub gridView1_CustomRowCellEdit(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Grid.CustomRowCellEditEventArgs)
            If e.Column.FieldName = "LargeImage" Then
                e.RepositoryItem = New DevExpress.XtraEditors.Repository.RepositoryItemPictureEdit()
            End If
        End Sub

        Private Sub OnFormLoad(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            gridControl1.DataSource = DataHelper.CreateTable()
        End Sub
    End Class
End Namespace